helga_jeopardy.py

```diff
@@ -267,6 +361,170 @@ def scores(client, channel, nick, alltime=False):
 
         rank += 1
 
+def fetch_categories():
+    random_cat = random.randint(1,18412)
+    cats_resp = requests.get('{}categories?count=6&offset={}'.format(api_endpoint, random_cat))
+    y=0
+    cat_dict={}
+    for x in ["cat1", "cat2", "cat3", "cat4", "cat5", "cat6"]:
+        cat_dict[x] = requests.get('{}category?id={}'.format(api_endpoint,cats_resp.json()[y]["id"]))
+        y += 1
+    return cat_dict
+
+def setup_new_game(client, channel, nick, message, cmd, args, mongo_db=db.jeopardy):
+    cat_dict = fetch_categories()
+    print cat_dict
+
+    for key, value in cat_dict.items():
+        category = cat_dict[key].json()
+        if len(category['clues']) < 5 or any(['question' not in q or not q['question'] for q in category['clues']]):
+            client.msg(channel, "Category response was malformed, requesting again.")
+            cat_dict = reactor.callLater(1, setup_new_game, client, channel, nick, message, cmd, args)
+            return
+
+    game = {
+    'channel': channel,
+    'game_active': True,
+    'game_started': False,
+    'game_host': nick,
+    }
+    for category_num in range(1, 7):
+        category_name = 'cat{}'.format(category_num)
+        category = cat_dict[category_name].json()
+        game[category_name] = {'category': category['title']}
+        set_value = 200
+        for i in range(5):
+            clue_name = 'clue{}'.format(i + 1)
+            question = category['clues'][i]
+            game_question = {
+                'question': question['question'],
+                'answer': question['answer'],
+                'value': set_value,
+                'id': question['id'],
+                'active': True,
+            }
+            game[category_name][clue_name] = game_question
+            set_value += 200
+
+    game_id = db.jeopardy.insert(game)
+    client.msg(channel, "New game created. to join: !j game join")
+    return
+
+def start_new_game(nick, new_game, client, channel, mongo_db=db.jeopardy):
+    current_players = []
+    for i in new_game["players"]:
+        current_players.append(i)
+
+    if nick not in current_players:
+        client.msg(channel, "you have not joined the game lobby. try: !j game join")
+        return
+
+    for player in current_players:
+        mongo_db.update({
+            'game_active': True,
+            'game_started': False,
+        }, {
+            '$set': {
+                player: 0,
+           }
+        })
```

Comment:
You don't need to copy the list of players since you're not modifying it.

```suggestion
    if nick not in new_game['players']:
        client.msg(channel, 'You have not joined the game lobby. Try: !j game join')
        return

    for player in new_game['players']:
        mongo_db.update({
            'game_active': True,
            'game_started': False,
        }, {
            '$set': {
                player: 0,
           }
        })
```